classes/Domain/Services/ResetEmailer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace OpenCFP\Domain\Services;
+
+class ResetEmailer
```

Comment:
Wouldn't gain us anything as far as switching to another mailing implementation. Hrm... can't find words over text so may just drop it and think some more. I'll try to explain better.

This is 100% the right abstraction for the job of "sending an account reset email". If I need to send a mail, in general (say as a result of some action in the API), I might not want to fulfill a dependency on Twig to send a plain-text email that a client application was registered. I probably wouldn't use this service and that's okay.

What *would* happen is that I would have to (at some point) grab references to `Swift_Mailer` in terms of the next best Service to do the "job of today". I may use that work as an opportunity to refactor the duplication I'm anticipating today.

However, I guess what I'm getting at more than anything is that the job of sending a account password reset notification is prompted by user action and could be orchestrated (with the right abstractions) as a controller action without a need to implement an service explicitly purposed for that one job.

Either way, it doesn't really matter so much I suppose. Just a point of discussion. I'd like to eventually move more and more towards something like:

```php
class AccountController extends Controller 
{
    public function processPasswordResetRequest(Request $request)
    {
        // Load up the user.
        $user = $this->identityProvider->getCurrentUser();
        $event = $user->resetPassword();

        $this->dispatcher->send($event);
    }
}

class AccountNotifier 
{
    public function whenPasswordWasReset(PasswordWasReset $event)
    {
        // Do the needful with whatever mailing abstraction / message construction.
        $this->mailer->send($event->getUserEmail(), $event->getMessage());  // Or similar...
        
        // ...
    }
}
```

All that said. I fully understand that this PR also represents a refactor for the purposes of demonstrating introducing tests into a legacy project, so I can appreciate that. We can continue down the path you've cut so far! :smile: Thanks!